src/main/java/org/maxgamer/quickshop/EnvironmentChecker.java

```diff
@@ -20,16 +20,22 @@
 package org.maxgamer.quickshop;
 
 import lombok.Getter;
+import org.bukkit.Material;
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.YamlConfiguration;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
 import org.jetbrains.annotations.NotNull;
 import org.maxgamer.quickshop.util.GameVersion;
 import org.maxgamer.quickshop.util.ReflectFactory;
 import org.maxgamer.quickshop.util.Util;
 
-public class RuntimeCatcher {
+public class EnvironmentChecker {
     @Getter
     private final GameVersion gameVersion;
+    private boolean hasCustomItemSavingBug = false;
 
-    public RuntimeCatcher(@NotNull QuickShop plugin) {
+    public EnvironmentChecker(@NotNull QuickShop plugin) {
```

Comment:
Codacy found an issue: [The constructor 'EnvironmentChecker(QuickShop)' has an NPath complexity of 864, current threshold is 200](https://app.codacy.com/gh/Ghost-chu/QuickShop-Reremake/pullRequest?prid=6383074)